### PR TITLE
[#268] Google/Facebook source documentation

### PR DESCRIPTION
**Pull request by paulodiniz**
Fixes https://github.com/airyhq/airy/issues/268

### CODE REVIEW

File: docs/docs/sources/facebook.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+title: Facebook Source
+sidebar_label: Facebook Source
```
Comment (chrismatix) [inline docs/docs/sources/facebook.md:3]: The directory is already called `sources`, so this tree will read like

sources
  facebook source
  google source

So I think you can drop

Comment (lucapette) [reply docs/docs/sources/facebook.md:3]: must drop :)

File: docs/docs/sources/facebook.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+## Introduction
```
Comment (lucapette) [inline docs/docs/sources/facebook.md:6]: I'd drop (it's kind of redundant)

File: docs/docs/sources/facebook.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+The Facebook source provides a channel of communication between your managed Facebook page and your running instance of the airy core.
+
+Like for any other source you must connect a channel first using the [channels
+connection endpoint](api/http.md#connecting-channels) and setting the `source`
+field in the request payload to `facebook`. The token parameter must be the token provided by Facebook.
```
Comment (chrismatix) [inline docs/docs/sources/facebook.md:12]: More explicitly the `user access token`

**Review by chrismatix — COMMENTED** [review]
This PR should also get rid of the TODOs in the `docs/docs/http.md` (line 394 and 458)

File: docs/docs/sources/facebook.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: Facebook Source
+sidebar_label: Facebook Source
+---
+
+## Introduction
+
+The Facebook source provides a channel of communication between your managed Facebook page and your running instance of the airy core.
```
Comment (lucapette) [inline docs/docs/sources/facebook.md:8]:
```suggestion
The Facebook source provides a channel of communication between a Facebook page and your running instance of the Airy Core Platform.
```

File: docs/docs/sources/facebook.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+## Events
+
+After the verification process, Facebook will start sending events to your local instance.
+
+```json5
+POST /facebook
+```
+
+Those events will be processed and turned into messages.
```
Comment (lucapette) [inline docs/docs/sources/facebook.md:30]:
```suggestion
Once the verification process has been completed, Facebook will immediately start sending events to your Airy Core Platform instance.
```

File: docs/docs/sources/google.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+---
+title: Google Source
+sidebar_label: Google Source
```
Comment (lucapette) [inline docs/docs/sources/google.md:3]: we should drop source

File: docs/docs/sources/google.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+## Introduction
```
Comment (lucapette) [inline docs/docs/sources/google.md:6]: I would drop

File: docs/docs/sources/google.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+The Google source provides a channel of communication between your Google Business Location and your running instance of the airy core.
```
Comment (lucapette) [inline docs/docs/sources/google.md:8]:
```suggestion
The Google source provides a channel of communication between your Google Business Location and your running instance of the Airy Core Platform.
```

File: docs/docs/sources/facebook.md (hunk 3 above)
Comment (lucapette) [inline docs/docs/sources/facebook.md:12]:
```suggestion
When connecting a channel via the [/channels.connect](api/http.md#connecting-channels), you must set the `source` field `facebook` and the `token` field to a valid Facebook user access token
```

File: docs/docs/sources/google.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+## Configuration
+
+First step is to copy the Google Service Account file provided by Google to `infrastructure/airy.conf` as a one line string
+
+```
+GOOGLE_SA_FILE=<CONTENT OF THE FILE>
+```
+
+As a security measure, every request sent by Google is signed and veryfied against your partner key.
```
Comment (lucapette) [inline docs/docs/sources/google.md:22]:
```suggestion
As a security measure, every request sent by Google is signed and verified against your partner key.
```

File: docs/docs/sources/google.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+After fetching it from the Google Business account, set the environment variable `GOOGLE_PARTNER_KEY` with the value.
```
Comment (lucapette) [inline docs/docs/sources/google.md:23]:
```suggestion
You must also set the environment variable `GOOGLE_PARTNER_KEY` to your partner key
```

File: docs/docs/sources/google.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+## Events
+
+```json5
+POST /google
+```
+
+Those events will be processed and turned into messages.
```
Comment (lucapette) [inline docs/docs/sources/google.md:31]:
```suggestion
Once the verification process has been completed, Google will immediately start sending events to your Airy Core Platform instance.
```

**Review by chrismatix — COMMENTED** [review]
There are two more TODO sections in `docs/docs/http.md` (line 394 and 485) that need to be merged into these docs.

**Review by lucapette — CHANGES_REQUESTED** [review]
Requesting changes because in some parts we refer to the wrong source

File: docs/docs/api/http.md [hunk 12]
```diff
@@ -391,41 +391,10 @@ to connect the channel.
...
-This request requires an authentication `token`, which has a different meaning for each source:
-
-- `facebook` The page access token
-- `google` leave empty. To allow authentication you must provide a [Google service account key file](https://developers.google.com/business-communications/business-messages/guides/quickstarts/prerequisite-setup) in your runtime configuration.
-
-The meaning of `source_channel_id` is also different per source
-
-- `facebook` The page id
-- `google` The id of your Google Business Message [agent](https://developers.google.com/business-communications/business-messages/reference/business-communications/rest/v1/brands.agents#Agent). 
+Connecting a channel is source specific, so you must read the source documentation for the correct payload.
```
Comment (lucapette) [inline docs/docs/api/http.md:394]:
```suggestion
Connecting a channel is source specific by nature, refer to the relevant documentation for the correct payload:
```

File: docs/docs/sources/facebook.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+sidebar_label: Facebook
+---
+
+The Facebook source provides a channel of communication between a Facebook page and your running instance of the Airy Core Platform.
+
+## Configuration
+
+For Facebook to start sending events to your local instance, it must first verify your instance with a challenge.
+The only configuration needed for that to happen is to set the environment variable `FB_WEBHOOK_SECRET` with the provided token
+by Facebook.
```
Comment (lucapette) [inline docs/docs/sources/facebook.md:12]:
```suggestion
To verify your Facebook webhook integration, you must set the environment variable `FB_WEBHOOK_SECRET` to the same value you supply in the webhook configuration on the Facebook integration.
```

File: docs/docs/sources/facebook.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+## Connecting a channel
+
+Connects a Facebook page to an user account.
+
+```
+POST /channels.connect
+```
+
+- `source` MUST be `facebook`
```
Comment (lucapette) [inline docs/docs/sources/facebook.md:26]:
```suggestion
- `source` *must* be `facebook`
```

File: docs/docs/sources/google.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+## Connecting a channel
+
+Connects a Google Business Account to an user account.
+
+```
+POST /channels.connect
+```
+
+- `source` MUST be `google`
```
Comment (lucapette) [inline docs/docs/sources/google.md:29]:
```suggestion
- `source` *must* be `google`
```

File: docs/docs/sources/google.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+  "image_url": "https://example.org/custom-image.jpg" // optional
+}
+```
+
+**Sample Response**
+
+```json5
+{
+	"id": "channel-uuid-1",
+    "name": "My custom name for this page",
```
Comment (lucapette) [inline docs/docs/sources/google.md:47]:
```suggestion
    "id": "channel-uuid-1",
    "name": "My custom name for this location",
```

File: docs/docs/sources/google.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    "image_url": "https://example.org/custom-image.jpg",
```
Comment (lucapette) [inline docs/docs/sources/google.md:48]:
```suggestion
    "image_url": "https://example.com/custom-image.jpg",
```

File: docs/docs/sources/google.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+
+- `source` MUST be `google`
+- `source_channel_id` The id of your Google Business Message [agent](https://developers.google.com/business-communications/business-messages/reference/business-communications/rest/v1/brands.agents#Agent). 
+- `token` leave empty. To allow authentication you must provide a [Google service account key file](https://developers.google.com/business-communications/business-messages/guides/quickstarts/prerequisite-setup) in your runtime configuration.
+
+```json5
+{
+  "source": "google",
+  "source_channel_id": "gbm-id",
+  "name": "My custom name for this page", // optional
```
Comment (lucapette) [inline docs/docs/sources/google.md:37]:
```suggestion
  "name": "My custom name for this location", // optional
```

File: docs/docs/sources/google.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+  "image_url": "https://example.org/custom-image.jpg" // optional
```
Comment (lucapette) [inline docs/docs/sources/google.md:38]:
```suggestion
  "image_url": "https://example.com/custom-image.jpg" // optional
```

**Review by lucapette — CHANGES_REQUESTED** [review]
Last pass! I promise 😇

File: docs/docs/sources/google.md (hunk 18 above)
Comment (chrismatix) [inline docs/docs/sources/google.md:37]: Ah name here is actually not optional, because we do not fetch the agent name. Sorry missed this! 😬

**Review by lucapette — APPROVED** [review]
:shipit:

**Pull request merged**